main/autogen/subclasses.cc

```diff
@@ -63,7 +63,7 @@ optional<Subclasses::Map> Subclasses::listAllSubclasses(core::Context ctx, Parse
                                             "::", [&ctx](const core::NameRef &nm) -> string { return nm.show(ctx); }));
 
         auto &mapEntry = out[parentName];
-        mapEntry.entries.insert(make_pair(childName, defn.data(pf).type));
+        mapEntry.entries.insert(make_tuple(childName, defn.data(pf).type, pf.path));
```

Comment:
To allow for better readability, given we have one more extra field now, I'd be more comfortable with this being a struct rather than a tuple, like `struct DescendantsMapEntry` or something similar. Not sure if the core Sorbet folks agree.

Reply:
This core Sorbet folk has a strong preference for named structs over against pairs and tuples.

I think the main challenge with converting to structs will be writing the necessary overloads to work with abseil's hash tables.

Reply:
The entries are stored in an unordered set so it does unfortunately.

Reply:
Ah

Reply:
You should be able to cargo-cult from https://github.com/sorbet/sorbet/blob/40a70d0acb9f5f5435808e12098693a129add54f/core/LocalVariable.h#L76-L78 for converting this into a struct.

You also -- I think -- don't have to hash the path, nor do you have to store the path as a string (assuming that the path is coming from `GlobalState`'s storage), which should help with the slowdown issues.